server/channels/store/sqlstore/channel_store_categories.go

```diff
@@ -437,15 +442,38 @@ func (s SqlChannelStore) completePopulatingCategoryChannels(category *model.Side
 	return result, nil
 }
 
-func (s SqlChannelStore) completePopulatingCategoryChannelsT(db dbSelecter, category *model.SidebarCategoryWithChannels) (*model.SidebarCategoryWithChannels, error) {
+func (s SqlChannelStore) completePopulatingCategoryChannelsT(db dbSelecter, category *model.SidebarCategoryWithChannels, appsCategoryEnabled bool) (*model.SidebarCategoryWithChannels, error) {
 	if category.Type == model.SidebarCategoryCustom || category.Type == model.SidebarCategoryFavorites {
 		return category, nil
 	}
 
 	var channelTypeFilter sq.Sqlizer
-	if category.Type == model.SidebarCategoryDirectMessages {
-		// any DM/GM channels that aren't in any category should be returned as part of the Direct Messages category
-		channelTypeFilter = sq.Eq{"Channels.Type": []model.ChannelType{model.ChannelTypeDirect, model.ChannelTypeGroup}}
+	if category.Type == model.SidebarCategoryApps || category.Type == model.SidebarCategoryDirectMessages {
+		botChannelsIDs := []string{}
+		if appsCategoryEnabled {
+			botChannels, err := s.SqlStore.stores.channel.GetBotChannelsByUser(category.UserId, store.ChannelSearchOpts{})
+			if err != nil {
+				return nil, err
+			}
+
+			for _, botChannel := range botChannels {
+				botChannelsIDs = append(botChannelsIDs, botChannel.Id)
+			}
+		}
+
+		if category.Type == model.SidebarCategoryApps {
+			// any DM channels with Bots that aren't in any category should be returned as part of the Apps category
+			channelTypeFilter = sq.And{
+				sq.Eq{"Channels.Type": model.ChannelTypeDirect},
+				sq.Eq{"Channels.Id": botChannelsIDs},
+			}
+		} else if category.Type == model.SidebarCategoryDirectMessages {
+			// any DM/GM channels with Users that aren't in any category should be returned as part of the Direct Messages category
+			channelTypeFilter = sq.And{
+				sq.Eq{"Channels.Type": []model.ChannelType{model.ChannelTypeDirect, model.ChannelTypeGroup}},
+				sq.NotEq{"Channels.Id": botChannelsIDs},
+			}
```

Comment:
What happens if the user purposely moved a bot channel to the DMs? It seems this would not return it?

Reply:
What if a user tries to change this directly via the API?

Reply:
Channels in the System Sidebar Categories (channels, dm, apps) are not really stored, but queried on request.
The enforce in the webapp side is just to make this experience consistent.